dev/SwipeControl/SwipeControl_TestUI/SwipeControlClearPage.xaml.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Windows.UI.Xaml;
+
+namespace MUXControlsTestApp
+{
+    /// <summary>
+    /// Test page used for clearing existing SwipeControls
+    /// </summary>
+    public sealed partial class SwipeControlClearPage : TestPage
+    {
+        public SwipeControlClearPage()
+        {
+            this.InitializeComponent();
+            SwipeItemsChildSum.Text = (DefaultSwipeItemsHorizontal.Count + DefaultSwipeItemsVertical.Count).ToString();
+        }
+
+        public void AddSwipeItemsButton_Click(object sender, RoutedEventArgs e)
+        {
+            DefaultSwipeItemsHorizontal.Clear();
+            DefaultSwipeItemsVertical.Clear();
+
+            DefaultSwipeItemsHorizontal.Mode = Microsoft.UI.Xaml.Controls.SwipeMode.Reveal;
+            DefaultSwipeItemsHorizontal.Add(DefaultSwipeItemHorizontal);
+
+            // Throws exception:
+            //"The parameter is incorrect. This SwipeControl is horizontal and can not have vertical items."
+            //DefaultSwipeItemsVertical.Mode = Microsoft.UI.Xaml.Controls.SwipeMode.Reveal;
+            //DefaultSwipeItemsVertical.Add(DefaultSwipeItemVertical);
```

Comment:
Unfortunately whether or not the swipe control is horizontal or vertical is locked in during on apply template, and if there are neither vertical or horizontal items at that time we default to being horizontal, so you are getting this error because you are attempting to add vertical items to a horizontal swipe control.  To avoid this you can add an item from markup.

Reply:
Oh okay, I did not know that. Thank you for your suggestion. I added them in markup and everything works now correctly!